src/context/internal.rs

```diff
@@ -10,17 +10,30 @@ use object::class::ClassMap;
 use result::NeonResult;
 use super::ModuleContext;
 
+#[cfg(feature = "legacy-runtime")]
 #[repr(C)]
 #[derive(Clone, Copy)]
-pub struct Isolate(*mut raw::Isolate);
+pub struct Env(raw::Isolate);
```

Comment:
Most of the `neon` code only uses the higher level `Env`, but `runtime` methods accept the result of `to_raw` which may either be an `Isolate` or `node_env`. This pattern lets us trust the type checker when implementing `runtime` with `napi`.